src/GraphQL/Authorization/AuthorizationExtensions.cs

```diff
@@ -63,6 +80,76 @@ public static TMetadataProvider AuthorizeWith<TMetadataProvider>(this TMetadataP
             return provider;
         }
 
+        /// <inheritdoc cref="AuthorizeWithPolicy{TMetadataProvider}(TMetadataProvider, string)"/>
+        [Obsolete("Please use AuthorizeWithPolicy instead.")]
+        public static TMetadataProvider AuthorizeWith<TMetadataProvider>(this TMetadataProvider provider, string policy)
+            where TMetadataProvider : IProvideMetadata
+            => AuthorizeWithPolicy(provider, policy);
+
+        /// <summary>
+        /// Adds authorization role(s) to the specified metadata provider. Roles should
+        /// be comma-separated and role names will be trimmed. If the underlying field already
+        /// contains a role with the same name, then it will not be added twice.
+        /// </summary>
+        /// <typeparam name="TMetadataProvider"> The type of metadata provider. Generics are used here to
+        /// let compiler infer the returning type to allow methods chaining.
+        /// </typeparam>
+        /// <param name="provider">
+        /// Metadata provider. This can be an instance of <see cref="GraphType"/>,
+        /// <see cref="FieldType"/>, <see cref="Schema"/> or others.
+        /// </param>
+        /// <param name="roles"> Comma-separated list of authorization role name(s). </param>
+        /// <returns> The reference to the specified <paramref name="provider"/>. </returns>
+        public static TMetadataProvider AuthorizeWithRoles<TMetadataProvider>(this TMetadataProvider provider, string roles)
+            where TMetadataProvider : IProvideMetadata
+        {
+            if (roles == null)
+                throw new ArgumentNullException(nameof(roles));
+
+            var list = GetRoles(provider) ?? new List<string>();
+
+            foreach (var role in roles.Split(','))
+            {
+                var role2 = role.Trim();
+                if (!list.Contains(role2))
+                    list.Add(role2);
+            }
+
+            provider.Metadata[ROLE_KEY] = list;
+            return provider;
+        }
+
+        /// <summary>
+        /// Adds authorization role(s) to the specified metadata provider.  If the underlying field already
+        /// contains a role with the same name, then it will not be added twice.
+        /// </summary>
+        /// <typeparam name="TMetadataProvider"> The type of metadata provider. Generics are used here to
+        /// let compiler infer the returning type to allow methods chaining.
+        /// </typeparam>
+        /// <param name="provider">
+        /// Metadata provider. This can be an instance of <see cref="GraphType"/>,
+        /// <see cref="FieldType"/>, <see cref="Schema"/> or others.
+        /// </param>
+        /// <param name="roles"> List of authorization role name(s). </param>
+        /// <returns> The reference to the specified <paramref name="provider"/>. </returns>
+        public static TMetadataProvider AuthorizeWithRoles<TMetadataProvider>(this TMetadataProvider provider, params string[] roles)
```

Comment:
I think it is worth to add calls without `params` arg in tests to verify no-op for all such methods like `field.AuthorizeWithRoles()`

Reply:
I'm not sure I follow.  The tests run both overloads of AuthorizeWithRoles -- all 6 new methods.  They also run the old policy methods and the obsolete method.  What did I miss?

Reply:
Oh I think I know what you mean.

Reply:
Done